### DISCUSSION TITLE
.ply or .pcd files and trajopt

### DISCUSSION THREAD

**Discussion by savolla** [tesseract-robotics/trajopt#316] [Q&A]
Can Trajopt operate on Point Cloud data? I have a very big pointcloud data and converting it to a mesh file takes too long. I want to know if I can just use the raw pointcloud data instead of .stl or .dae files.

**Comment by Levi-Armstrong** [comment]
It does. Tesseract support creating the tesseract_geometry::Octree object from a point cloud. I recommend using the tesseract_geometry::Octree::prune or setting the constructor parameter prune to true. This is a custom prune which looks at adjacent cells and if occupied it grows the box so fewer collision objects are created.
